### ISSUE TITLE
Cannot use 

### ISSUE THREAD

**Issue by TubboHere** [victornpb/undiscord#679]
### Undiscord Version

5.2.3

### Browser

131.0.6778.86 (Official Build) (x86_64)

### Extension

Tampermonkey

### Reproduction steps

Did the usual steps filled in all the needed fiellds

### Actual results

Error searching messages, API responded with status 401!
	{"message":"401: Unauthorized","code":0}
CoreException	{}

### Expected results

deleted messages

### Bug affects

- [X] DMs
- [ ] DMs (group)
- [ ] Channels
- [ ] Channel (NSFW)

### Additional information & file uploads

![image](https://github.com/user-attachments/assets/04a2c12b-495c-438d-afac-879a99c29a8a)

**Comment by vajdao** [comment]
same here! @victornpb

**Comment by An0n-00** [comment]
401 means unauthorized. try later again. you probably hit the rate limit for discord. (or there was/is a misconfiguration, or you (your account) is not allowed to delete messages from the server/channel)

**Comment by MouadQ793** [comment]
@An0n-00 I disagree, I have same error, I tried in multiple servers, even my own and it doesn't work it shows same error and I am p sure I have the right to delete messages

**Comment by Hark64** [comment]
I tried on multiple servers and channels and I get the exact same error in the screenshot.

**Comment by An0n-00** [comment]
hmm. lol strang. i think it is a wrong error msg in that case.

**Comment by PesterMess** [comment]
It seems like this issue may be related to Discord's recent changes in how links function when navigating between DMs and server channels. I hope Victornpb can review and address these changes soon.

**Comment by gaveitatry2** [comment]
If it's just a 401 error, I fixed this back in Aug 2024. All you had to do was scroll down the Issues page to around eight issues before yours. 401 errors are caused because UnDiscord is using the user's Discord password as the authentication token instead of the actual authentication token.

https://github.com/victornpb/undiscord/issues/653

**Comment by Hark64** [comment]
I guess then I have an issue where it does not automatically get the auth token. But even when I manually put it in it still does not work and I get a 400 error instead.

![Screenshot 2025-01-11 131744](https://github.com/user-attachments/assets/e5a28b75-9db2-48e3-964d-c4c0154cec03)

**Comment by gaveitatry2** [comment]
I just got done explaining that we fixed the UnDiscord userscript and we put it on Greasy Fork. So you can just install the fixed version. No need to manually add authentication tokens. That was the old method we used before fixing the userscript itself.

**Comment by Hark64** [comment]
Hey I'm really sorry for the confusion. I didn't mean to upset you.

I did download undiscord on greasyfork but when I searched I found two scripts called undiscord, undiscord remade, undiscord patched, and undiscord v2. But only the one updated in 2023 had more than 1k downloads, it has 474k downloads. That's also the only one with victornpb as the author and was the only one that received updates after launch so I assumed that was the legit version.

Even so I still get the 400 error with the others. Again, I'm really sorry for upsetting you, I'm just confused as to what is going on and looking for a little help because I love the tool you guys have made.

**Comment by gaveitatry2** [comment]
You weren't supposed to "search on Greasy Fork" for the userscript. My userscript is unlisted but I provided a direct link to it. All you had to do was click my Github link to get to the Greasy Fork userscript install link.

**Comment by Hark64** [comment]
Sorry man. Geez...

Nothing that you said made that clear. All you had to do was say to click on your profile instead of saying greasyfork. If you don't want to share it or are annoyed by github issues like this just don't respond and keep your unlisted.

**Comment by gaveitatry2** [comment]
You're still crying? After all I did for you, you still haven't installed it or thanked me yet? Sheesh. Why do I bother? I get that I was being a complete d head. But I think everyone on this thread is slow. That PesterMess guy? I showed him how to manually add authentication tokens. I even made a YouTube video for him. Never thanked me. Then I fixed the userscript and @'d him like a dozen times. Still didn't try the userscript or thank me. That was six months ago. Everyone on this "Cannot use" thread is slow. All they want to is be stupid, helpless, and cry. Instead of posting "does not work" and "cannot use" threads they literally only had to scroll down a fraction of the page and see "SOLUTION FOR 401 ERRORS". But these people put forth 1% effort or they really are just that slow. I feel sorry for you all. I'm spoon feeding you the solutions and you are still stuck, still ungrateful, still whining.

**Comment by Hark64** [comment]
Wow, I'm stunned. I don't even know what to say to that.

**Comment by PesterMess** [comment]
What did I just miss? My email shows someone complaining about us not expressing gratitude, and it’s honestly laughable. I hope they take a moment to reflect on their actions instead of trying to trash talk others on GitHub. Thanks to Victornpb for removing them!

**Comment by Hark64** [comment]
Hey sorry to bombard your email. gaveitatry2 got upset I couldn't figure out where to find his link to his greasyfork version and somehow he got emotional. I honestly don't know what happened either.

**Comment by PesterMess** [comment]
I get it. Some people really need to take a break and get some fresh air. It’s easy to share a link in a comment! Could you send me the link too? I'm getting tired of waiting for a fix.

**Comment by Hark64** [comment]
Yeah for sure. I still get a 400 error with it though.
https://greasyfork.org/en/scripts/502561-undiscord

**Comment by PesterMess** [comment]
Then it's not worth my time, lol.

**Comment by Hark64** [comment]
Yeah I guess so, lol.

**Comment by LavenderAndChamomile** [comment]
I'm also getting this issue....
